docs/source/distributed.rst

```diff
@@ -293,6 +293,33 @@ check whether the process group has already been initialized use :func:`torch.di
 
 .. autofunction:: get_world_size
 
+Shutdown
+--------
+
+It is important to clean up resources on exit by calling :func:`destroy_process_group`.
+
+The simplest pattern to follow is to destroy every process group and backend by calling
+:func:`destroy_process_group()` with the default value of None for the `group` argument, at a
+point in the training script where communications are no longer needed, usually near the
+end of main().  The call should be made once per trainer-process, not at the outer
+process-launcher level.
+
+if :func:`destroy_process_group` is not called, especially when there are multiple process-groups
```

Comment:
maybe add something like " if :func:`destroy_process_group` is not called or not called  by all ranks in a pg within the timeout duration". 

we saw some cases in which some ranks (usually rank 0) calls abort just too 'late'.